presto-main/src/main/java/com/facebook/presto/operator/aggregation/differentialentropy/DifferentialEntropyAggregation.java

```diff
@@ -0,0 +1,129 @@
+        if (state.getStrategy() == null) {
+            state.setStrategy(StateSerializer.create(
+                    size, sample, weight, requestedMethod, min, max));
+        }
+        StateStrategy strategy = state.getStrategy();
+        StateSerializer.validate(strategy, requestedMethod);
+        strategy.validateParameters(size, sample, weight, min, max);
+        strategy.add(sample, weight);
+    }
+
+    @InputFunction
+    public static void input(
+            @AggregationState State state,
+            @SqlType(StandardTypes.BIGINT) long size,
+            @SqlType(StandardTypes.DOUBLE) double sample,
+            @SqlType(StandardTypes.DOUBLE) double weight,
+            @SqlType(StandardTypes.VARCHAR) Slice method)
+    {
+        String requestedMethod = method.toStringUtf8();
+        if (state.getStrategy() == null) {
+            state.setStrategy(StateSerializer.create(
+                    size, sample, weight, requestedMethod));
+        }
+        StateStrategy strategy = state.getStrategy();
+        StateSerializer.validate(strategy, requestedMethod);
+        strategy.add(sample, weight);
+    }
+
+    @InputFunction
+    public static void input(
+            @AggregationState State state,
+            @SqlType(StandardTypes.BIGINT) long size,
+            @SqlType(StandardTypes.DOUBLE) double sample,
+            @SqlType(StandardTypes.DOUBLE) double weight)
+    {
+        if (state.getStrategy() == null) {
+            state.setStrategy(StateSerializer.create(size, sample, weight));
+        }
+        StateStrategy strategy = state.getStrategy();
+        StateSerializer.validate(strategy);
+        strategy.add(sample, weight);
+    }
+
+    @InputFunction
+    public static void input(
+            @AggregationState State state,
+            @SqlType(StandardTypes.BIGINT) long size,
+            @SqlType(StandardTypes.DOUBLE) double sample)
+    {
+        if (state.getStrategy() == null) {
+            state.setStrategy(StateSerializer.create(size, sample));
+        }
+        StateStrategy strategy = state.getStrategy();
+        StateSerializer.validate(strategy);
+        strategy.add(sample, 1.0);
+    }
+
+    @CombineFunction
+    public static void combine(
+            @AggregationState State state,
+            @AggregationState State otherState)
+    {
+        StateStrategy strategy = state.getStrategy();
+        StateStrategy otherStrategy = otherState.getStrategy();
+        if (strategy == null && otherStrategy != null) {
+            state.setStrategy(otherStrategy.clone());
+            return;
+        }
+        if (strategy != null && otherStrategy != null) {
+            StateSerializer.combine(strategy, otherStrategy);
+        }
+    }
+
+    @OutputFunction("double")
+    public static void output(@AggregationState State state, BlockBuilder out)
+    {
+        StateStrategy strategy = state.getStrategy();
+        DOUBLE.writeDouble(
```

Comment:
put arguments on the same line: `DOUBLE.writeDouble(out, strategy == null ? Double.NaN : strategy.calculateEntropy());`